DiscordBot/Commands/EventCommands.cs

```diff
@@ -217,6 +219,72 @@ public async Task EditEvent(CommandContext context, InteractivityModule interact
             await EditEventField(context, interactivity, eventKey.Value, editField, eventEmbed);
         }
 
+        private static async Task CreateSignupSheet(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the event key? (use the ``list`` option to find out)");
+            var eventKey = await GetUserIntResponse(context, interactivity);
+            if (eventKey == null)
+            {
+                return;
+            }
+
+            var eventEmbed = await GetEventEmbed(context, eventKey.Value);
+            if (eventEmbed == null)
+            {
+                return;
+            }
+
+            await context.RespondAsync($"{context.Member.Mention} - start signups for this event? - (``yes``/``no``)", embed: eventEmbed);
+            var confirmationResponse = await GetUserConfirmation(context, interactivity);
+            if (confirmationResponse == null || confirmationResponse == false)
+            {
+                return;
+            }
+
+            await SendSignupMessage(context, eventKey.Value);
+        }
+
+        private static async Task SendSignupMessage(CommandContext context, int eventKey)
+        { 
+            var eventsSheetsService = context.Dependencies.GetDependency<IEventsSheetsService>();
+            var discordEvent = await eventsSheetsService.GetEventAsync(eventKey);
+
+            var signupEmbed = new DiscordEmbedBuilder
+            {
+                Title = $"{discordEvent.Name} - {discordEvent.Time:ddd dd MMM yyyy @ h:mm tt}",
+                Description = discordEvent.Description,
+                Footer = new DiscordEmbedBuilder.EmbedFooter
+                {
+                    Text = $"event key: {eventKey}"
+                }
+            };
+
+            var userResponseDictionary = await eventsSheetsService.GetSignupsByUserAsync(eventKey);
+
+            var usersField = string.Join("\n", userResponseDictionary.Select(
+                response => $"<@{response.Key}>"));
```

Comment:
If you're going to split this over several lines then each argument to the function needs to be on a new line, otherwise it's a bit tricky to read. For example:
```
var usersField = string.Join(
  "\n",
  userResponseDictionary.Select(response => $"<@{response.Key}>")
);
```
or
```
var usersField = string.Join(
  "\n", 
  userResponseDictionary.Select(
    response => $"<@{response.Key}>"
  )
);
```
Does that make sense? 🙂